debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/converters/sink/types/bson/BytesType.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.mongodb.converters.sink.types.bson;
+
+import java.nio.ByteBuffer;
+
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.errors.DataException;
+import org.bson.BsonBinary;
+import org.bson.BsonValue;
+
+public class BytesType extends AbstractBsonType {
+
+    public BytesType() {
+        super(Schema.BYTES_SCHEMA);
+    }
+
+    @Override
+    public BsonValue toBson(final Object data) {
```

Comment:
We can test Java pattern matching on this connector In Debezium 3.x when we swtich to Java 17.

Reply:
Agreed! +1 definitely!